### PR TITLE
Doxygen style

### PR DESCRIPTION
**Pull request by wmamills**
This is the Doxygen Style work and the updates to the Sphinx theme to match.
I suggest we accept this as a baseline and we can tweak later (before end of October or later depending on how things go.)
I think this is better than what we had and is pretty close.

I will make an issue for the remain style issues as I see them.

(This PR won't style open-amp as that repo has not accepted the style enable PR yet)

### CODE REVIEW

File: _doxygen/openamp/footer.html [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+<!--END GENERATE_TREEVIEW-->
+<!--BEGIN !GENERATE_TREEVIEW-->
+<hr class="footer"/><address class="footer"><small>
+$generatedby&#160;<a href="https://www.doxygen.org/index.html"><img class="footer" src="$relpath^doxygen.svg" width="104" height="31" alt="doxygen"/></a> $doxygenversion
+</small></address>
+<!--END !GENERATE_TREEVIEW-->
+<script type="text/javascript">
+  $(function() {
+    toggleButton = document.createElement('doxygen-awesome-dark-mode-toggle')
+    toggleButton.title = "Toggle Light/Dark Mode"
```
Comment (arnopo) [inline _doxygen/openamp/footer.html]: I can not see the button in https://openamp--26.org.readthedocs.build/en/26/ page
But perhaps need to be merged to see it?

**Review by arnopo — APPROVED** [review]
LGTM with a question

**Comment by wmamills** [comment]
@arnopo it only for the Doxygen right now not the Sphinx.
And the open-amp doxygen is not styled right now because the support PR has not been merged.
So in the this PR you will only see it on the libmetal docs.
I checked and it is there.
https://openamp--26.org.readthedocs.build/en/26/doxygen/libmetal/index.htm

I am working on one more fix.  Put the correct version number on the Doxygen page.
Give me 1 hour please

**Comment by wmamills** [comment]
Ok added one more patch to this PR.
After it renders you should see the correct version name in libmetal

**Comment by wmamills** [comment]
@nathalie-ckc @tammyleino Can I get a 2nd approval from one of you?
Again I would like to get this in as the baseline and we can take further improvements

Now that all the supporting PRs are in I have rebased and pushed again.
The doc's built above are basicly what we will have.
Here is the link just to make it easy:
https://openamp--26.org.readthedocs.build/en/26/

Issue #27 has the todo list.
Issue #28 has one todo item that is bigger.

Feel free to add more issues or edit #27 to add small items.

**Review by tammyleino — APPROVED** [review]

**Review by nathalie-ckc — APPROVED** [review]
Approving based on clicking through the preview at https://openamp--26.org.readthedocs.build/en/26/

**Pull request merged**